frontend/src/components/panels/Document.vue

```diff
@@ -330,6 +334,10 @@ export default defineComponent({
 			const updateData = responseData as ExportDocument;
 			if (updateData) this.download("canvas.svg", updateData.document);
 		});
+		registerResponseHandler(ResponseType.SaveDocument, (responseData: Response) => {
+			const saveData = responseData as SaveDocument;
+			if (saveData) this.download("canvas.graphite", saveData.document);
```

Comment:
I think the last thing we need to change is setting the file name to the document (tab) name. So it would, for example, be `Untitled Document 3.graphite` if you tried saving the third tab you opened. I think you'll need to plumb that through as string data along with this `SaveDocument` message in the enum data.